internal/fileio/fileio.go

```diff
@@ -85,3 +111,43 @@ func ensureDirectory(d string) error {
 	}
 	return nil
 }
+
+func mergeConfigs(paths []string) (map[string]interface{}, error) {
+	base := map[string]interface{}{}
+
+	for _, halPath := range paths {
+		currentMap := map[string]interface{}{}
+
+		data, err := ioutil.ReadFile(halPath)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read %q: %w", halPath, err)
+		}
+
+		if err := yaml.Unmarshal(data, &currentMap); err != nil {
+			return nil, fmt.Errorf("unable to parse %q: %w", halPath, err)
+		}
+
+		base = mergeMaps(base, currentMap)
+	}
+
+	return base, nil
+}
+
+func mergeMaps(a, b map[string]interface{}) map[string]interface{} {
```

Comment:
I guess this is a fairly simple function, so maybe the benefit of pulling it in from a library might be questionable...but is this something that there might be a library for?  In the absence of that some tests on it would probably be good.

Reply:
I actually borrowed this from Helm's merge implementation, let me know if this requires any additional citing.